From: =?utf-8?q?Fabian_Gr=C3=BCnbichler?= <git@fabian.gruenbichler.email>
Date: Sat, 20 Sep 2025 09:58:14 +0200
Subject: Revert "rustc_target: Add target features for LoongArch v1.1"

This reverts commit d989bf5bbe511769d23fb43361f59d6a6ac76bc7.

Description: this requires LLVM 20..
---
 compiler/rustc_target/src/target_features.rs | 5 -----
 tests/ui/check-cfg/target_feature.stderr     | 5 -----
 2 files changed, 10 deletions(-)

diff --git a/compiler/rustc_target/src/target_features.rs b/compiler/rustc_target/src/target_features.rs
index 3eea1e0..7434967 100644
--- a/compiler/rustc_target/src/target_features.rs
+++ b/compiler/rustc_target/src/target_features.rs
@@ -699,18 +699,13 @@ static CSKY_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
 static LOONGARCH_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
     // tidy-alphabetical-start
     ("d", Stable, &["f"]),
-    ("div32", Unstable(sym::loongarch_target_feature), &[]),
     ("f", Stable, &[]),
     ("frecipe", Stable, &[]),
-    ("lam-bh", Unstable(sym::loongarch_target_feature), &[]),
-    ("lamcas", Unstable(sym::loongarch_target_feature), &[]),
     ("lasx", Stable, &["lsx"]),
     ("lbt", Stable, &[]),
-    ("ld-seq-sa", Unstable(sym::loongarch_target_feature), &[]),
     ("lsx", Stable, &["d"]),
     ("lvz", Stable, &[]),
     ("relax", Unstable(sym::loongarch_target_feature), &[]),
-    ("scq", Unstable(sym::loongarch_target_feature), &[]),
     ("ual", Unstable(sym::loongarch_target_feature), &[]),
     // tidy-alphabetical-end
 ];
diff --git a/tests/ui/check-cfg/target_feature.stderr b/tests/ui/check-cfg/target_feature.stderr
index f422919..28a0625 100644
--- a/tests/ui/check-cfg/target_feature.stderr
+++ b/tests/ui/check-cfg/target_feature.stderr
@@ -70,7 +70,6 @@ LL |     cfg!(target_feature = "_UNEXPECTED_VALUE");
 `d32`
 `deflate-conversion`
 `dit`
-`div32`
 `doloop`
 `dotprod`
 `dpb`
@@ -144,11 +143,8 @@ LL |     cfg!(target_feature = "_UNEXPECTED_VALUE");
 `jsconv`
 `kl`
 `lahfsahf`
-`lam-bh`
-`lamcas`
 `lasx`
 `lbt`
-`ld-seq-sa`
 `leoncasa`
 `lor`
 `lse`
@@ -213,7 +209,6 @@ LL |     cfg!(target_feature = "_UNEXPECTED_VALUE");
 `relaxed-simd`
 `rtm`
 `sb`
-`scq`
 `sha`
 `sha2`
 `sha3`
